blockstack_client/data.py

```diff
@@ -170,29 +147,40 @@ def delete_mutable_data_version(conf, data_id):
     Return False if not
     """
 
-    if conf is None:
-        conf = config.get_config()
+    conf = config.get_config() if conf is None else conf
 
     if conf is None:
-        log.warning("No config found; cannot store version for '%s'" % data_id)
+        msg = 'No config found; cannot store version for "{}"'
+        log.warning(msg.format(data_id))
         return False
 
     metadata_dir = conf['metadata']
     if not os.path.isdir(metadata_dir):
-        log.warning("No metadata directory found; cannot store version of '%s'" % data_id)
+        msg = 'No metadata directory found; cannot store version of "{}"'
+        log.warning(msg.format(data_id))
         return False
 
-    serialized_data_id = data_id.replace("/", "\x2f").replace('\0', "\\0")
-    version_file_path = os.path.join(metadata_dir, serialized_data_id + ".ver")
+    serialized_data_id = data_id.replace('/', '\x2f').replace('\0', '\\0')
+    version_file_path = os.path.join(
+        metadata_dir, '{}.ver'.format(serialized_data_id)
+    )
 
     try:
         os.unlink(version_file_path)
         return True
-
-    except Exception, e:
+    except Exception as e:
         # failed for whatever reason
-        log.warn("Failed to remove version file '%s'" % (version_file_path))
-        return False
+        msg = 'Failed to remove version file "{}"'
+        log.warn(msg.format(version_file_path))
+
+    return False
+
+
+def is_legacy_profile(user_zonefile):
```

Comment:
I like that this check is now consolidated :)  However, some clarification on the nature of these checks is needed.  The reason for these checks isn't to determine if the profile has a legacy format; the reason is to verify that the data whose hash is on the blockchain corresponds to a well-formed zonefile/profile pair.  Both checks are needed because a specially-crafted legacy profile can also pass as a well-formed zonefile, and we only want to support storage operations if we're certain that the given information is *actually* a zonefile (i.e. if a user has a legacy profile that also looks like a zonefile, then (s)he should first run the `migrate` command to switch over to a zonefile that is guaranteed to not be a legacy profile).

I would consider calling this method `is_obsolete_zonefile`.